client/block.c

```diff
@@ -1941,16 +1941,36 @@ void xdag_block_finish()
 	pthread_mutex_lock(&block_mutex);
 }
 
-int xdag_get_block_info(xdag_hash_t hash, void *data, int (*callback)(void*, int, xdag_hash_t, xdag_amount_t, xtime_t, const char *))
+int xdag_get_block_info(xdag_hash_t hash, void *info, int (*info_callback)(void*, int, xdag_hash_t, xdag_amount_t, xtime_t, const char *),
+						void *links, int (*links_callback)(void*, const char *, xdag_hash_t, xdag_amount_t))
 {
 	pthread_mutex_lock(&block_mutex);
 	struct block_internal *bi = block_by_hash(hash);
 	pthread_mutex_unlock(&block_mutex);
 
-	if(callback && bi) {
-		return callback(data, bi->flags & ~BI_OURS,  bi->hash, bi->amount, bi->time, get_remark(bi));
+	if(info_callback && bi) {
+		info_callback(info, bi->flags & ~BI_OURS,  bi->hash, bi->amount, bi->time, get_remark(bi));
 	}
 
+	if(links_callback && bi) {
+		int flags;
+		struct block_internal *ref;
```

Comment:
you can join declaration of variables with assigning.

Reply:
it is to lock the mutex the minimum time, he declare and define the variables outside the mutex area

Reply:
See answer 2: https://stackoverflow.com/questions/27729930/is-declaration-of-variables-expensive

Reply:
Yes, it's very fast, but still takes some time, also depends on optimization.

Reply:
> 
> 
> Yes, it's very fast, but still takes some time, also depends on optimization.

@xrdavies Depending on size of local variables compiler just reserves some space in the stack. But declaration of variables does not take additional time.

Reply:
so wherever they are, it is the same as they are on the top :+1:

Reply:
> > Yes, it's very fast, but still takes some time, also depends on optimization.
> 
> @xrdavies Depending on size of local variables compiler just reserves some space in the stack. But declaration of variables does not take additional time.

Yes, you are correct.